packages/app/src/cli/services/app-logs/poll-app-logs.ts

```diff
@@ -0,0 +1,97 @@
+import {writeAppLogsToFile} from './write-app-logs.js'
+import {partnersFqdn} from '@shopify/cli-kit/node/context/fqdn'
+import {fetch} from '@shopify/cli-kit/node/http'
+import {Writable} from 'stream'
+
+const POLLING_INTERVAL_MS = 450
+const ONE_MILLION = 1000000
+
+const generateFetchAppLogUrl = async (cursor?: string) => {
+  const fqdn = await partnersFqdn()
+  const url = `https://${fqdn}/app_logs/poll`
+  return url + (cursor ? `?cursor=${cursor}` : '')
+}
+
+export interface AppEventData {
+  shop_id: number
+  api_client_id: number
+  payload: string
+  event_type: string
+  cursor: string
+  status: 'success' | 'failure'
+  log_timestamp: string
+}
+
+export const pollAppLogs = async ({
+  stdout,
+  appLogsFetchInput: {jwtToken, cursor},
+  apiKey,
+}: {
+  stdout: Writable
+  appLogsFetchInput: {jwtToken: string; cursor?: string}
+  apiKey: string
+}) => {
+  const url = await generateFetchAppLogUrl(cursor)
+  const response = await fetch(url, {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${jwtToken}`,
+    },
+  })
+
+  if (!response.ok) {
+    // We should add some exponential backoff here to not spam partners
+
+    const responseText = await response.text()
+    throw new Error(`Error while fetching: ${responseText}`)
```

Comment:
💭 what does this look like to the end user? Curious if the dev would know what this error relates to